### PR TITLE
Dev mofijul

### PR DESCRIPTION
**Pull request by MOFIJULHAQUE**
**Right sidebar Design Updated...**

### CODE REVIEW

File: src/components/RightSideBar/CurrencyConverterComponent/CurrencyInput.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import PropTypes from "prop-types";
+import "./currencyInput.css";
+
+function CurrencyInput(props) {
+  return (
+    <div className="group">
+      <input
+        type="text"
+        value={props.amount}
+        onChange={(ev) => props.onAmountChange(ev.target.value)}
```
Comment (Souvikdeb2612) [inline src/components/RightSideBar/CurrencyConverterComponent/CurrencyInput.jsx]: Create separate functions and call them here

File: src/components/RightSideBar/CurrencyConverterComponent/CurrencyInput.jsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      />
+      <select
+        value={props.currency}
+        onChange={(ev) => props.onCurrencyChange(ev.target.value)}
```
Comment (Souvikdeb2612) [inline src/components/RightSideBar/CurrencyConverterComponent/CurrencyInput.jsx]: same as above

File: src/components/RightSideBar/CurrencyConverterComponent/CurrencyInput.jsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import PropTypes from "prop-types";
+import "./currencyInput.css";
+
+function CurrencyInput(props) {
```
Comment (Souvikdeb2612) [inline src/components/RightSideBar/CurrencyConverterComponent/CurrencyInput.jsx]: destructure props here

File: src/components/RightSideBar/Reaction.jsx [hunk 4]
```diff
@@ -1,12 +1,261 @@
...
+  const input = useRef();
+
+  const date = new Date();
+  const hours = date.getHours();
+  const seconds = date.getSeconds();
+  const day = date.getDay();
+  const month = date.getMonth();
+  const year = date.getFullYear();
+
+  const days = [
```
Comment (Souvikdeb2612) [inline src/components/RightSideBar/Reaction.jsx]: create constants folder and store all data there.

File: src/components/RightSideBar/Reaction.jsx [hunk 5]
```diff
@@ -1,12 +1,261 @@
...
+    "Sunday",
+    "Monday",
+    "Tuesday",
+    "Wednesday",
+    "Thursday",
+    "Friday",
+    "Saturday",
+  ];
+  const months = [
```
Comment (Souvikdeb2612) [inline src/components/RightSideBar/Reaction.jsx]: same here

File: src/components/RightSideBar/Reaction.jsx [hunk 6]
```diff
@@ -1,12 +1,261 @@
...
+      setTimeout(() => {
+        getBotMessage.innerText = "Bye, have a nice day";
+        inputRef.value = ""; // clear the input
+      }, 2000);
+      setTimeout(() => {
+        status.innerText = "Not active";
+        status.style.color = "red";
+      }, 5000);
+    }
+    let thanks = [
```
Comment (Souvikdeb2612) [inline src/components/RightSideBar/Reaction.jsx:108]: What are these messages and what's the use?

File: src/components/RightSideBar/Reaction.jsx [hunk 7]
```diff
@@ -1,12 +1,261 @@
...
+    let words8 = new RegExp(nameAsk);
+    if (words8.test(document.querySelector("#input").value)) {
+      getBotMessage.innerText = "Typing...";
+      setTimeout(() => {
+        getBotMessage.innerText = "My name is Bot";
+        inputRef.value = ""; // clear the input
+      }, 2000);
+    }
+
+    let owner = [
```
Comment (Souvikdeb2612) [inline src/components/RightSideBar/Reaction.jsx:170]: What are all these?

**Review by Souvikdeb2612 — CHANGES_REQUESTED** [review]

File: src/components/RightSideBar/Coverter.jsx [hunk 8]
```diff
@@ -15,9 +15,8 @@ export const Coverter=()=> {
   const [currency1, setCurrency1] = useState('USD');
   const [currency2, setCurrency2] = useState('EUR');
   const [rates, setRates] = useState([]);
-
   useEffect(() => {
-    axios.get('https://api.apilayer.com/fixer/latest?base=USD&apikey=pcKgTvH5OIq8SqqdQYH9O5QV99hoUvE9')
+    axios.get('https://api.apilayer.com/fixer/latest?base=USD&apikey=CrH3fXURs0j3YngQZjop9V8NqmkQgd5L')    //API limit exceeds
```
Comment (Souvikdeb2612) [inline src/components/RightSideBar/Coverter.jsx]: store url separately and use the variable here

File: src/components/RightSideBar/CurrencyConverterComponent/CurrencyInput.jsx [hunk 9]
```diff
@@ -1,19 +1,20 @@
...
+function CurrencyInput({amount,currency,currencies,onAmountChange,onCurrencyChange}) {
+  
   return (
     <div className="group">
       <input
         type="text"
-        value={props.amount}
-        onChange={(ev) => props.onAmountChange(ev.target.value)}
+        value={amount}
+        onChange={(ev) => onAmountChange(ev.target.value)}
```
Comment (Souvikdeb2612) [inline src/components/RightSideBar/CurrencyConverterComponent/CurrencyInput.jsx]: create a function and call here

File: src/components/RightSideBar/CurrencyConverterComponent/CurrencyInput.jsx [hunk 10]
```diff
@@ -1,19 +1,20 @@
...
       />
       <select
-        value={props.currency}
-        onChange={(ev) => props.onCurrencyChange(ev.target.value)}
+        value={currency}
+        onChange={(ev) => onCurrencyChange(ev.target.value)}
```
Comment (Souvikdeb2612) [inline src/components/RightSideBar/CurrencyConverterComponent/CurrencyInput.jsx]: same as above

File: src/pages/Policy.jsx [hunk 11]
```diff
@@ -1,9 +1,160 @@
-import React from 'react'
+import React from "react";
+import image1 from "../Images/privacy (1).png";
```
Comment (Souvikdeb2612) [inline src/pages/Policy.jsx:3]: all static image files should be exported separately from an index file and finally destructure and import specific files wherever required

File: src/pages/Policy.jsx [hunk 12]
```diff
@@ -1,9 +1,160 @@
...
+          </div>
+        </div>
+
+        <div className="privacy_policy">
+          <div>
+            <img src={image3} alt="image3" />
+          </div>
+          <div>
+            <p>
+              Lorem ipsum dolor sit amet consectetur adipisicing elit. Rerum
```
Comment (Souvikdeb2612) [inline src/pages/Policy.jsx]: Store Lorem Ipsum text separately in constant folder and import wherever required

**Review by Souvikdeb2612 — CHANGES_REQUESTED** [review]

File: src/components/RightSideBar/CurrencyConverterComponent/CurrencyInput.jsx (hunk 1 above)
Comment (MOFIJULHAQUE) [reply src/components/RightSideBar/CurrencyConverterComponent/CurrencyInput.jsx]: Sir I tried here but not done

File: src/components/RightSideBar/CurrencyConverterComponent/CurrencyInput.jsx (hunk 9 above)
Comment (MOFIJULHAQUE) [reply src/components/RightSideBar/CurrencyConverterComponent/CurrencyInput.jsx]: Sir I tried here but not done

File: src/components/Login_signUp/Signup.jsx [hunk 13]
```diff
@@ -0,0 +1,190 @@
+import React, { useState } from "react";
+
+import { Button, Form, Input } from "antd";
+
+import "./loginSignup.css";
+
+// Ant deign
+const MainLoginSignup = () => {
+  const onFinish = (values) => {
+    console.log("Success:", values);
```
Comment (Souvikdeb2612) [inline src/components/Login_signUp/Signup.jsx:10]: remove/comment console statements

File: src/components/Login_signUp/Signup.jsx [hunk 14]
```diff
@@ -0,0 +1,190 @@
...
+  };
+  const onFinishFailed = (errorInfo) => {
+    console.log("Failed:", errorInfo);
```
Comment (Souvikdeb2612) [inline src/components/Login_signUp/Signup.jsx:13]: same as above

**Review by Souvikdeb2612 — CHANGES_REQUESTED** [review]

**Review by Souvikdeb2612 — APPROVED** [review]

**Pull request merged**